java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -91,4 +89,118 @@ protected static ConnectionManager buildConnectionManager(ChannelHandler channel
+     * @return Response <code>Object</code>
+     */
+    protected Object handleObjectResponse(Response response) {
+        // convert protobuf response into Object
+        return new BaseCommandResponseResolver(RedisValueResolver::valueFromPointer).apply(response);
+    }
+
+    /**
+     * Checks that the Response is empty.
+     *
+     * @return An empty response
+     */
+    protected Ok handleOkResponse(Response response) {
+        Object value = handleObjectResponse(response);
+        if (value == Ok.INSTANCE) {
+            return Ok.INSTANCE;
+        }
+        throw new RedisException(
+                "Unexpected return type from Redis: got "
+                        + value.getClass().getSimpleName()
+                        + " expected simple-string Ok");
+    }
+
+    /**
+     * Extracts the response value from the Redis response and either throws an exception or returns
+     * the value as a <code>String</code>.
+     *
+     * @param response Redis protobuf message
+     * @return Response as a <code>String</code>
+     */
+    protected String handleStringResponse(Response response) {
+        Object value = handleObjectResponse(response);
+        if (value instanceof String || value == null) {
+            return (String) value;
+        }
+        if (value instanceof Ok) {
+            return Ok.INSTANCE.toString();
+        }
+        throw new RedisException(
+                "Unexpected return type from Redis: got "
+                        + value.getClass().getSimpleName()
+                        + " expected String");
+    }
+
+    /**
+     * Extracts the response value from the Redis response and either throws an exception or returns
+     * the value as an <code>Object[]</code>.
+     *
+     * @param response Redis protobuf message
+     * @return Response as an <code>Object[]</code>
+     */
+    protected Object[] handleArrayResponse(Response response) {
+        Object value = handleObjectResponse(response);
+        if (value instanceof Object[]) {
+            return (Object[]) value;
+        }
+        throw new RedisException(
+                "Unexpected return type from Redis: got "
+                        + value.getClass().getSimpleName()
+                        + " expected Object[]");
+    }
+
+    /**
+     * Extracts the response value from the Redis response and either throws an exception or returns
+     * the value as a <code>Map</code>.
+     *
+     * @param response Redis protobuf message
+     * @return Response as a <code>Map</code>.
+     */
+    @SuppressWarnings("unchecked")
+    protected Map<Object, Object> handleMapResponse(Response response) {
+        Object value = handleObjectResponse(response);
+        if (value instanceof Map) {
+            return (Map<Object, Object>) value;
+        }
+        throw new RedisException(
+                "Unexpected return type from Redis: got "
+                        + value.getClass().getSimpleName()
```

Comment:
```suggestion
                        + (value == null ? "null" : value.getClass().getSimpleName())
```

Reply:
or we could just return null in all these cases...?

Reply:
btw - good catch